### PR TITLE
MINOR: Updating docs for enabling optimizations

### PR DESCRIPTION
**Pull request by bbejeck**
Companion changes from https://github.com/apache/kafka/pull/7099  in AK repo.

### CODE REVIEW

**Comment by bbejeck** [comment]
ping @vvcephei, @ableegoldman, @cadonna, @abbccdda

**Comment by bbejeck** [comment]
\cc @JimGalasyn @londoncalling

File: 23/streams/upgrade-guide.html [hunk 1]
```diff
@@ -80,6 +80,14 @@ <h3><a id="streams_api_changes_230" href="#streams_api_changes_230">Streams API
         The new public interfaces <code>inMemoryWindowStore()</code> and <code>inMemorySessionStore()</code> are added to <code>Stores</code> and provide the built-in in-memory window or session store.
     </p>
 
+    <p>
+        As of 2.3.0 we've updated how to turn on optimizations. Now to enable optimizations, you need to do two things.
```
Comment (JimGalasyn) [inline 23/streams/upgrade-guide.html]:
```suggestion
        As of 2.3.0 we've updated how to turn on optimizations. Now, to enable optimizations, you need to do two things.
```

File: 23/streams/upgrade-guide.html [hunk 2]
```diff
@@ -80,6 +80,14 @@ <h3><a id="streams_api_changes_230" href="#streams_api_changes_230">Streams API
...
+        First, add this line to your properties <code>properties.setProperty(StreamsConfig.TOPOLOGY_OPTIMIZATION, StreamsConfig.OPTIMIZE);</code>, as you have done before.
```
Comment (JimGalasyn) [inline 23/streams/upgrade-guide.html]:
```suggestion
        First, add this line to your properties, <code>properties.setProperty(StreamsConfig.TOPOLOGY_OPTIMIZATION, StreamsConfig.OPTIMIZE);</code>, as you have done before.
```

File: 23/streams/upgrade-guide.html [hunk 3]
```diff
@@ -80,6 +80,14 @@ <h3><a id="streams_api_changes_230" href="#streams_api_changes_230">Streams API
...
+        Second, when constructing your <code>KafkaStreams</code> instance, you'll need to pass your configuration properties when building your
+        topology via the overloaded <code>StreamsBuilder.build(Properties)</code> method.
```
Comment (JimGalasyn) [inline 23/streams/upgrade-guide.html]:
```suggestion
        topology by using the overloaded <code>StreamsBuilder.build(Properties)</code> method.
```

File: 23/streams/upgrade-guide.html [hunk 4]
```diff
@@ -80,6 +80,14 @@ <h3><a id="streams_api_changes_230" href="#streams_api_changes_230">Streams API
...
+        For example <code>KafkaStreams myStream = new KafkaStreams(streamsBuilder.build(properties), properties)</code>.
```
Comment (JimGalasyn) [inline 23/streams/upgrade-guide.html]:
```suggestion
        For example, <code>KafkaStreams myStream = new KafkaStreams(streamsBuilder.build(properties), properties)</code>.
```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a couple of suggestions.

**Review by mjsax — APPROVED** [review]

**Pull request merged**